math/src/polynomial.rs

```diff
@@ -158,6 +145,25 @@ impl<F: IsField> Polynomial<FieldElement<F>> {
         self.coefficients.pop();
     }
 
+    pub fn ruffini_division<L>(&self, b: &FieldElement<L>) -> Polynomial<FieldElement<L>>
+    where
+        L: IsField,
+        F: IsSubFieldOf<L>,
+    {
+        if let Some(c) = self.coefficients.last() {
+            let mut c = c.clone().to_extension();
+            let mut coefficients = Vec::with_capacity(self.degree());
+            for coeff in self.coefficients.iter().rev().skip(1) {
+                coefficients.push(c.clone());
+                c = coeff + c * b;
+            }
+            coefficients = coefficients.into_iter().rev().collect();
```

Comment:
You can [reverse](https://doc.rust-lang.org/stable/std/vec/struct.Vec.html#method.reverse) the vector in-place:
```suggestion
            coefficients.reverse();
```